testapp/src/main/java/com/rakuten/tech/mobile/testapp/ui/userdata/QASettingsActivity.kt

```diff
@@ -104,11 +104,62 @@ class QASettingsActivity : BaseActivity() {
         binding.switchUniqueIdError.setOnCheckedChangeListener { _, isChecked ->
             binding.edtUniqueIdError.isEnabled = isChecked
         }
+        binding.btnClearMiniAppSecureStorage.setOnClickListener{
+            if (!binding.clearStorageForMiniAppId.text.isNullOrEmpty()) {
+                clearSecureStorageForMiniApp(binding.clearStorageForMiniAppId.text.toString())
+            } else {
+                Toast.makeText(
+                    this@QASettingsActivity,
+                    "MiniApp ID cannot be empty to clear MiniApp storage",
+                    Toast.LENGTH_LONG
+                ).show()
+            }
+        }
         binding.btnClearAllSecureStorage.setOnClickListener {
             clearAllSecureStorage()
         }
     }
 
+    private fun clearSecureStorageForMiniApp(miniAppId: String) {
+        if (!doesMiniAppIdExist(miniAppId)) {
+            Toast.makeText(
+                this@QASettingsActivity,
+                "Could not find the MiniApp to clear the secured storage!",
+                Toast.LENGTH_LONG
+            ).show()
+        } else {
+            val dialogClickListener =
+                DialogInterface.OnClickListener { dialog, which ->
+                    when (which) {
+                        DialogInterface.BUTTON_POSITIVE -> {
+                            miniApp.clearSecureStorage(miniAppId)
+                            Toast.makeText(
```

Comment:
if `MiniApp.clearSecureStorage` could return a boolean after finished the clearing, then we could confirm to show an UI like this. Otherwise, this Toast message is redundant when the storage is not cleared successfully.

Reply:
@gopi-karmakar i agree with this comment, if you add some sort of callback or just return a `boolean` to actually detemine the db is successfully deleted may be a good idea to have, in `link` this might come handy if something wrong during deleting the db.

Reply:
Addressed